crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals_is_not.md

```diff
@@ -0,0 +1,37 @@
+# Narrowing for `is not` conditionals
+
+## `is not None`
+
+The type guard removes `None` from the union type:
+
+```py
+x = None if flag else 1
+
+if x is not None:
+    reveal_type(x)  # revealed: Literal[1]
+
+reveal_type(x)  # revealed: None | Literal[1]
+```
+
+## `is not` for other singleton types
+
+```py
+x = True if flag else False
+reveal_type(x)  # revealed: bool
+
+if x is not False:
+    # TODO the following should be `Literal[True]`
+    reveal_type(x)  # revealed: bool & ~Literal[False]
+```
+
+## `is not` for non-singleton types
+
+Non-singleton types should *not* narrow the type:
+
+```py
+x = [1]
+y = [1]
+
+if x is not y:
+    reveal_type(x)  # revealed: list
```

Comment:
Ideally, this should be `list[int]`, but I guess that is being tracked elsewhere, so I didn't add a TODO comment.

Reply:
Hmm... I guess I would weakly prefer adding a `TODO: generics` comment here, so that it's obvious to readers of the test that we are not asserting this type because we believe that it is the best type we can possibly infer here, just because it's the best we can do right now

Reply:
Not important (definitely no need to change), but FWIW my preference would have been to just avoid using a generic type entirely here, and thus avoid the need for the TODO :) There are plenty of non-generic types (e.g. `int`) that would demonstrate precisely the same issue here.